crates/types/src/traits/election.rs

```diff
@@ -620,6 +620,8 @@ impl<
     ) -> (EncodedPublicKey, EncodedSignature) {
         let signature = TYPES::SignatureKey::sign(
             &self.private_key,
+            // TODO GG why create a VoteData::DA only to discard it immediately?
+            // Why not just have a `sign_commitment()` member?
```

Comment:
Because we want to distinguish the vote type when signing, so a signature on a no vote can't be used to verify a yes vote.

Reply:
But `get_commit()` just returns the underlying commitment regardless of the enum variant:
https://github.com/EspressoSystems/HotShot/blob/42d19bad389dab76c2ba073a6ffbd5daf5f5b08c/crates/types/src/traits/election.rs#L103-L111

Hmm. Seems it was me who made that change in #1777: https://github.com/EspressoSystems/HotShot/pull/1777/files

Ok, I need to fix this. I'll push more commits to this PR. Thanks for explaining!

Reply:
fixed in https://github.com/EspressoSystems/HotShot/pull/1864/commits/1fdc06abf724431e4f800565f75e9aef67810a96